RevenueCatUI/CustomerCenter/Data/CustomerCenterActionPreferencesViewModifier.swift

```diff
@@ -38,17 +39,28 @@ struct CustomerCenterActionPreferencesViewModifier: ViewModifier {
         content
             .onAppear {
                 // Set up direct binding to the state variables
-                actionWrapper.setRestoreStarted = { restoreStarted = true }
-                actionWrapper.setRestoreFailed = { restoreFailed = $0 as NSError }
-                actionWrapper.setRestoreCompleted = { restoreCompleted = $0 }
+                actionWrapper.setRestoreStarted = { 
+                    // Increment counter to create a new unique value
+                    // This guarantees preference change detection
+                    restoreCounter += 1
+                }
+                
+                actionWrapper.setRestoreFailed = { error in
+                    restoreFailed = error as NSError
```

Comment:
I believe the casting here is not necessary
```suggestion
                    restoreFailed = error
```